### DISCUSSION TITLE
Does Next JS send a default Content-security-policy header ?

### DISCUSSION THREAD

**Discussion by slevy85** [vercel/next.js#16728] [Help] [answered]
Hi,

I am trying to define my content-security-policy like this :  

```
async function headers() {
  return [
    {
      source: "/",
      headers: [
        {
          key: "Content-Security-Policy",
          value:
            "default-src * data: 'unsafe-inline' 'unsafe-eval'; worker-src * data: blob:; script-src * data: blob: 'unsafe-eval' 'unsafe-inline';",
        },
      ],
    },
  ];
}
```
I can see this header is correctly sent by the server, but my problem is that there are in fact two headers, this one, and an other one more restrictive : `default-src * data: 'unsafe-eval' 'unsafe-inline'`.

I don't know from where this second header is coming from and I can't access the server to check it out.

So my question is, does Next JS send me this header like a "default CSP" ?

Thank you for your help.

**Comment by ivan-kleshnin** [comment, marked as answer]
NextJS does not set any default CSP header. You should probably inspect your app closely.

**Reply by slevy85** [reply]
Thank you very much, and yes you are right, there was a NGinx server between the client and NextJS, the CSP was in NGinx config.
